core/capabilities/targets/write_target.go

```diff
@@ -108,20 +122,48 @@ func (cap *EvmWrite) Execute(ctx context.Context, request capabilities.Capabilit
 	if inputs.Report == nil {
 		// We received any empty report -- this means we should skip transmission.
 		cap.lggr.Debugw("Skipping empty report", "request", request)
-		callback := make(chan capabilities.CapabilityResponse)
-		go func() {
-			// TODO: cast tx.Error to Err (or Value to Value?)
-			callback <- capabilities.CapabilityResponse{
-				Value: nil,
-				Err:   nil,
-			}
-			close(callback)
-		}()
-		return callback, nil
+		return success(), nil
 	}
 
 	// TODO: validate encoded report is prefixed with workflowID and executionID that match the request meta
 
+	// Check whether value was already transmitted on chain
+	cr, err := evm.NewChainReaderService(ctx, cap.lggr, cap.chain.LogPoller(), cap.chain.Client(), relayevmtypes.ChainReaderConfig{
+		Contracts: map[string]relayevmtypes.ChainContractReader{
+			"forwarder": {
+				ContractABI: forwarder.KeystoneForwarderABI,
+				Configs: map[string]*relayevmtypes.ChainReaderDefinition{
+					"getTransmitter": {
+						ChainSpecificName: "getTransmitter",
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	var transmitter common.Address
+	cr.Bind(ctx, []commontypes.BoundContract{{
+		Address: config.ForwarderAddress().String(),
+		Name:    "forwarder",
+		// Pending: false, // ???
```

Comment:
@nolag I couldn't find any documentation on what this flag does

Reply:
If you are reading a contract method it doesn't really do anything.

Reply:
If you are reading a log event it translated to confirmations Finalized/Unfinalized